### PR TITLE
update istioctl installation guide based on new API

### PR DESCRIPTION
**Pull request by richardwxn**

### CODE REVIEW

File: content/en/docs/setup/install/istioctl/index.md [hunk 1]
```diff
@@ -388,17 +394,20 @@ which allows the following settings to be changed. Use this list to identify the
...
+The following example overlay file adjusts the resources and horizontal pod autoscaling
 settings for Pilot:
 
 {{< text yaml >}}
-apiVersion: install.istio.io/v1alpha2
-kind: IstioControlPlane
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
 spec:
-  trafficManagement:
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md]: When you delete this line, you need to unindent (by 2 spaces) all the following lines in the structure.

File: content/en/docs/setup/install/istioctl/index.md [hunk 2]
```diff
@@ -278,62 +278,76 @@ For example, the following command overrides the `pilot.traceSampling` Helm conf
...
+`nodeagent` |
+`galley` |
+`ingressGateways` |
+`egressGateways` |
+`cni` |
 
-In addition to the core Istio components, third-party addon features and components are also available. These can only
-be enabled and configured through the Helm pass-through API:
+In addition to the core Istio components, third-party addon components are also available. These can 
+be enabled and configured through the addonComponents spec of `IstioOperator` API or Helm pass-through API:
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md]:
```suggestion
be enabled and configured through the `addonComponents` spec of `IstioOperator` API or the Helm pass-through API:
```

File: content/en/docs/setup/install/istioctl/index.md [hunk 3]
```diff
@@ -278,62 +278,76 @@ For example, the following command overrides the `pilot.traceSampling` Helm conf
...
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+spec:
+  values:
+    grafana:
+      enabled: true
+{{< /text >}}
 
-### Configure the feature or component settings
+### Configure the component settings
```
Comment (frankbu) [inline content/en/docs/setup/install/istioctl/index.md]: Changing this title changes the id of the generated html section, so you need to fix the broken link.

Comment (frankbu) [reply content/en/docs/setup/install/istioctl/index.md]:
```
 - ./public/docs/setup/upgrade/istioctl-upgrade/index.html
  *  linking to /docs/setup/install/istioctl/#configure-the-feature-or-component-settings, but configure-the-feature-or-component-settings does not exist (line 65535)
     <a href="/docs/setup/install/istioctl/#configure-the-feature-or-component-settings">IstioControlPlane API Custom Resource Definition</a>
```

File: content/en/docs/setup/install/istioctl/index.md (hunk 2 above)
Comment (richardwxn) [reply content/en/docs/setup/install/istioctl/index.md]: done

File: content/en/docs/setup/install/istioctl/index.md (hunk 3 above)
Comment (richardwxn) [reply content/en/docs/setup/install/istioctl/index.md]: done

File: content/en/docs/setup/install/istioctl/index.md (hunk 1 above)
Comment (richardwxn) [reply content/en/docs/setup/install/istioctl/index.md]: done

**Pull request merged**